### ISSUE TITLE
Adding Fan Control

### ISSUE THREAD

**Issue by schartrand77** [NorthernMan54/homebridge-tcc#73]
Any way to implement fan control?
The TCC-Fan plugin breaks Homebridge now.

**Comment by NorthernMan54** [comment]
At the present time nope, will add it to the back log for a future enhancement.

If someone wants to submit a pull request with the feature I will accept it.

**Comment by macconnolly** [comment]
Could you share context here? At the present time what prevents it from working? Do you know which commit removed support so I can see how it was implemented previously? I’m happy to work on this! Just some guidance would be appreciated

**Comment by NorthernMan54** [comment]
It was never implemented, as I never change my fan settings

**Comment by NorthernMan54** [comment]
To implement this am looking at using a FanV2 Service added to the thermostat accessory for the fan.  So the fan be either a combo tile or a separate tile as needed.  But as HomeKit does not directly map Fan controls to the HoneyWell model, it will be a bit a kludge that will need to be worked thru.  I'm going to collect the various options here, so if you can share what Fan Modes your thermostat supports if they are not listed below.

HoneyWell Fan Options | [Active](https://developers.homebridge.io/#/characteristic/Active) Active/Inactive | [TargetFanState](https://developers.homebridge.io/#/characteristic/TargetFanState) Auto/Manual
-----|-----|------
On/Auto | |
On/Auto/Circulate | |


PS @donavanbecker

**Comment by donavanbecker** [comment]
![image](https://user-images.githubusercontent.com/9875439/87225792-fb5bfb80-c354-11ea-9bb4-e8aa4303700c.png)For on/auto/circulate, I was thinking if you turn on oscillating, that would change fan mode to manual and circulate. For Auto, if you have it be auto. And On is active.

**Comment by Freddicus** [comment]
I'm here from reddit! So the thing I didn't like about the existing tcc-fan plugin and using fan v1/v2 is that it obfuscates the four fan modes. I couldn't think of a good way to have them available, but you can [look at my code](https://github.com/Freddicus/homebridge-tcc-fan) if you want. Let me know what you think.

**Comment by NorthernMan54** [comment]
@Freddicus I'm leaning towards the FanV2 and am thinking to have the controls work as per the above tables.  Thru the various options, can get everything to work except follow Schedule.

**Comment by Freddicus** [comment]
@NorthernMan54  Cool! I won't encroach, then. :) I'll publish what I have for people who may want this alternative. I personally use Follow Schedule mode the most. Though now that I'm thinking I guess I could switch to just switch to schedule the thermostat with HomeKit and not use the scheduling functionality of TCC. I'll have to think about it.

Maybe one of the duplicated modes in the above table could be for Follow Schedule?

HoneyWell Fan Options | *Active* Active/Inactive | *TargetFanState* Auto/Manual | Honeywell Mode
------------------------|------------------------|--------------------------------|------------------
On/Auto/Circulate/FS | Active	| Manual	| Set fan to On
On/Auto/Circulate/FS	| Inactive	| Manual	| Set fan to circulate
On/Auto/Circulate/FS	| Active	| Auto	| Set fan to Auto
On/Auto/Circulate/FS	| Inactive	| Auto	| Set fan to FS

**Comment by NorthernMan54** [comment]
@Freddicus Let me see if I can get your suggestion to work

**Comment by ManBehindPlan** [comment]
I have the [RTH6580WF](https://www.honeywellhome.com/us/en/products/air/thermostats/wifi-thermostats/wi-fi-7-day-programmable-thermostat-rth6580wf1001-u1/) thermostat (most basic WIFI thermostat), and the [TCC Fan](https://github.com/hacctarr/homebridge-tcc-fan) works well for me (besides being chatty in the Homebridge logs, which I just put as a request for the plugin).

For me, the only two modes for fan are On and Auto, so it's the simplest approach.  When Home sets the fan to On, adjust the fan to ON in TCC.  When the Home sets the fan to off, adjust the fan in TCC to Auto.

**Comment by glitchykitten** [comment]
Hi friends.  I'd be willing to help with the fan support if I can get a bit of support and mentoring along the way.  (I'm an experienced programmer but this will be my first time in javascript and node.).  I have a vested interest in the outcome: I need to automate my fan settings between the three modes (on, circulate, auto) to get my HVAC to work efficiently.

I'm pretty new to, well, everything about this... so if I make any etiquette mistakes please just let me know.   :)

**Comment by NorthernMan54** [comment]
@glitchykitten We were all beginners at one point, so no issue with giving it a try.  I will be reviewing your pull request, so along as you are okay with some feedback it should be fine.

**Comment by glitchykitten** [comment]
Sounds good to me.  I'm ramping up now and will direct my questions to the Discord.  I'll run a high level plan by you before I code it up, and I'll look forward to feedback on the pull request.

**Comment by TeslaOwnerTips** [comment]
Was there any progress on adding fan controls? I'm interested in the circulate feature.

**Comment by glitchykitten** [comment]
No, I abandoned the project because I ended up re-doing my HVAC system and didn't need to automate fan controls anymore.  (And then after the debacle of homekit automation unreliability over the past few months, I actually moved all my HVAC automation back to the native thermostat scheduling and abandoned HomeKit automations entirely.)

**Comment by glitchykitten** [comment]
You'll need to talk to NorthernMan45 for more technical details since it's his plugin.  You can probably find him on the Discord.

**Comment by NorthernMan54** [comment]
@TeslaOwnerTips did you see this ? https://github.com/Freddicus/homebridge-tcc-fan

**Comment by jlg89** [comment]
So homebridge-tcc-fan works alongside homebridge-tcc, but if you have "too many" thermostats (not sure what the magic number is, but 13 is definitely too many), the two plugins hitting the TCC website will cause it to timeout quite reliably.

https://github.com/hacctarr/homebridge-tcc-fan/issues/4

My take on the fan control options is that most folks only need "on" or "auto," as setting a scheduled t-stat to "auto" effectively puts it back into "follow schedule" mode, and "on" is effectively the same as "circulate" for our purposes.

**Comment by Freddicus** [comment]
The four switches are now stateful and the plugin has been updated for modern home bridge: 
* https://github.com/Freddicus/homebridge-tcc-fan 
* https://www.npmjs.com/package/homebridge-tcc-fan-plus
